### PR TITLE
Fix terms for taste neurons of taste bristles

### PR DESCRIPTION
**Pull request by gouttegd**
This PR attempts the issue with the `S cell`, `W cell`, `L1 GRN`, and `L2 GRN` terms being defined as labellar-specific gustatory neurons even though they are used in the literature, and in some curation records, with a broader meaning valid for gustatory neurons in *any* taste bristle.

The approach (here for the `S cell` term) is basically to introduce an intermediate term between `sugar-sensing neuron` (referring to any such neuron, regardless of which structure it belongs to) and `sugar-sensing neuron of the labellum`, to represent sugar-sensing neurons that belong to (or rather, innervate) a taste bristle. That new term is the one carrying the EXACT `S cell` synonym, matching the current usage in the literature. The same approach is used for the other three types of gustatory neurons.

To account for the different ways the existing terms have been used in curation records:
* the current `S cell` term becomes the new `sugar-sensing neuron of taste bristle`, and `sugar-sensing neuron of the labellum` is a new term;
* likewise, the current `L2 GRN` term becomes the new `high salt-sensing neuron of taste bristle`, and `high salt-sensing neuron of the labellum` is a new term;
* the current `W cell` becomes `water-sensing neuron of the labellum` (so, keeping its labellar-specific meaning), and the parent `water-sensing neuron of taste bristle` is a new term;
* likewise for `L1 GRN`.

related to #1190

### CODE REVIEW

File: src/ontology/fbbt-edit.obo [hunk 1]
```diff
@@ -123372,6 +123418,30 @@ relationship: part_of FBbt:00100311 ! cortical zone of lymph gland primary lobe
 created_by: http://orcid.org/0000-0002-6095-8718
 creation_date: 2021-05-27T11:11:09Z
 
+[Term]
+id: FBbt:00058153
+name: high salt-sensing neuron of the labellum
+def: "A high salt-sensing neuron that innervates a short or long sensillum of the labellum (Montell, 2009)." [FlyBase:FBrf0208984]
```
Comment (Clare72) [inline src/ontology/fbbt-edit.obo:123424]: These are a subset of the labellar bitter-sensing neurons - the bitter neurons in S-a and S-b sensilla. From FBrf0229608:
>Bitter taste neurons in S-type sensilla also respond to high concentrations of salt

File: src/ontology/fbbt-edit.obo [hunk 2]
```diff
@@ -126355,58 +126425,62 @@ creation_date: 2010-02-17T03:56:47Z
...
 created_by: sr544
 creation_date: 2010-03-09T11:23:31Z
 
 [Term]
 id: FBbt:00100043
-name: L1 GRN
-def: "Gustatory receptor neuron of the short or long sensilla of the labellum that is responsive to low concentrations of salt (Montell, 2009)." [FlyBase:FBrf0208984]
-synonym: "classical salt neuron" RELATED [FlyBase:FBrf0137203]
+name: low salt-sensing neuron of the labellum
+def: "A low salt-sensing neuron that innervates a short or long sensillum of the labellum (Montell, 2009)." [FlyBase:FBrf0208984]
```
Comment (Clare72) [inline src/ontology/fbbt-edit.obo:126461]: looks like it is only long sensilla based on FBrf0229608, so this can be updated

**Review by Clare72 — CHANGES_REQUESTED** [review]
Probably worth looking at this (and maybe other) more recent reference(s) - FBrf0229608. It doesn't look as though low salt (appetitive) and water neurons exist outside of the labellum (so no need for parent classes). High salt is detected by some of the bitter neurons, rather than bitter being a subclass of a broader salt-detecting class.

File: src/ontology/fbbt-edit.obo [hunk 3]
```diff
@@ -123372,6 +123418,30 @@ relationship: part_of FBbt:00100311 ! cortical zone of lymph gland primary lobe
...
+subset: cur
+synonym: "L2 GRN" BROAD [FlyBase:FBrf0208984]
+intersection_of: FBbt:00100044 ! high salt-sensing neuron of taste bristle
+intersection_of: has_sensory_dendrite_in FBbt:00004162 ! labellar taste bristle
+created_by: http://orcid.org/0000-0002-6095-8718
+creation_date: 2021-08-05T12:09:40Z
+
+[Term]
+id: FBbt:00058154
+name: water-sensing neuron of taste bristle
```
[the remark this answers is not in the document]
Comment (gouttegd) [reply src/ontology/fbbt-edit.obo:123434]: They do according to at least one paper (FBrf0160792). Interesting that a recent review does not mention them, I'll have another look before going forward here.

File: src/ontology/fbbt-edit.obo [hunk 4]
```diff
@@ -122958,6 +122973,37 @@ is_a: FBbt:00003331 ! reductor muscle
...
+subset: cur
+intersection_of: FBbt:00005106 ! neuron
+intersection_of: capable_of GO:0001583 ! detection of chemical stimulus involved in sensory perception of salty taste
+intersection_of: has_sensory_dendrite_in FBbt:00005178 ! taste bristle
+created_by: http://orcid.org/0000-0002-6095-8718
+creation_date: 2021-08-05T11:32:14Z
+
+[Term]
+id: FBbt:00058112
+name: low salt-sensing neuron of taste bristle
```
[the remark this answers is not in the document]
Comment (gouttegd) [reply src/ontology/fbbt-edit.obo:122998]: Same as above, according to the same paper.

File: src/ontology/fbbt-edit.obo (hunk 3 above)
[the remark this answers is not in the document]
Comment (Clare72) [reply src/ontology/fbbt-edit.obo:123434]: Ah, it actually is in the figure, but not matched to the legend - looks like the functionality is not mapped to sensilla yet. Not clear if it is mapped to distict neurons yet either.

**Comment by gouttegd** [comment]
Closing here as another PR is on the way to fix this issue.

**Pull request closed without merging**